src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/MarkJavaObjects.cs

```diff
@@ -149,13 +149,55 @@ void PreserveAdapter (TypeDefinition type)
 			}
 		}
 
+		string TypeNameWithoutKey (string name)
+		{
+			var idx = name.IndexOf (", PublicKeyToken=");
+			if (idx > 0)
+				name = name.Substring (0, idx);
+
+			return name;
+		}
+
+		bool CheckInvokerType (TypeDefinition type, string name)
+		{
+			return TypeNameWithoutKey (name) == TypeNameWithoutKey ($"{ type.FullName}, { type.Module.Assembly.FullName}");
+		}
+
+		void PreserveInterfaceMethods (TypeDefinition type, TypeDefinition invoker)
+		{
+			foreach (var m in type.GetMethods ()) {
+				CustomAttribute registerAttr;
+				if (!m.TryGetRegisterAttribute (out registerAttr))
+					continue;
+
+				if (registerAttr.ConstructorArguments.Count != 3)
```

Comment:
Couldn't this use `TryGetRegisterMember()`?

Reply:
Yes, it can :-) Thanks. Indeed, I first thought I will need to use another constructor parameter and then forgot about it.